From 20927cf12803491a28438a62ba07facc81145f70 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 19 Apr 2022 14:33:05 +0300
Subject: [PATCH] [SW-0] gaudi3: remove die check for prstn configuration

This patch removes register configuration in prstn init based
on die id. Since HW takes care of adding offsets for die1
register when executed from die1, ROM can only make use of die0
registers for configurations.

Change-Id: I3ee35d1ff2c0757e696e7e9022cd19d5f4156cc7
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 242f5da..d728833 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -322,19 +322,11 @@
 
 static void arcboot_rom_prstn_init(void)
 {
-	if (arcboot_rom_global.die_id) {
-		regmap_writel(0, mmD1_PSOC_GLOBAL_CONF_PRSTN_MASK);
-		regmap_writel(D1_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG_EN_M,
-				mmD1_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
-		regmap_writel(D1_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG_EN_M,
-				mmD1_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
-	} else {
-		regmap_writel(0, mmD0_PSOC_GLOBAL_CONF_PRSTN_MASK);
-		regmap_writel(D0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG_EN_M,
-				mmD0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
-		regmap_writel(D0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG_EN_M,
-				mmD0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
-	}
+	regmap_writel(0, mmD0_PSOC_GLOBAL_CONF_PRSTN_MASK);
+	regmap_writel(D0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG_EN_M,
+			mmD0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
+	regmap_writel(D0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG_EN_M,
+			mmD0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
 }
 
 static void arcboot_rom_read_cold_flops(void)
